bin/tx-prover/src/prover.rs

```diff
@@ -1,49 +1,30 @@
 use alloc::{boxed::Box, string::ToString};
-use core::cell::RefCell;
 
 use miden_objects::transaction::{ProvenTransaction, TransactionWitness};
 use miden_tx::{TransactionProver, TransactionProverError};
 
-use crate::{generated::api_client::ApiClient, RemoteTransactionProverError};
+use crate::generated::api_client::ApiClient;
 
 // REMOTE TRANSACTION PROVER
 // ================================================================================================
 
 /// A [RemoteTransactionProver] is a transaction prover that sends witness data to a remote
 /// gRPC server and receives a proven transaction.
+///
+/// When compiled for the `wasm32-unknown-unknown` target, it uses the `tonic_web_wasm_client`
+/// transport. Otherwise, it uses the built-in `tonic::transport` for native platforms.
+/// The transport layer connection is established lazily when the first transaction is proven.
+
 #[derive(Clone)]
 pub struct RemoteTransactionProver {
-    #[cfg(target_arch = "wasm32")]
-    client: RefCell<ApiClient<tonic_web_wasm_client::Client>>,
-
-    #[cfg(not(target_arch = "wasm32"))]
-    client: RefCell<ApiClient<tonic::transport::Channel>>,
+    endpoint: String,
 }
```

Comment:
Would it make sense to keep the actual client here as well but as an `Option`? For example, something like this:
```rust
pub struct RemoteTransactionProver {
    endpoint: String,

    #[cfg(target_arch = "wasm32")]
    client: RefCell<Option<ApiClient<tonic_web_wasm_client::Client>>>,

    #[cfg(not(target_arch = "wasm32"))]
    client: RefCell<Option<ApiClient<tonic::transport::Channel>>>,
}
```
Then we could have something like `connect()` method that would return the actual client - e.g., something like this:
```rust
impl RemoteTransactionProver {
    #[cfg(target_arch = "wasm32")]
    fn connect(&self) -> &mut ApiClient<tonic_web_wasm_client::Client> {
        // create a new connection or if we've pervasively connected return an existing one
    }
}
```

Reply:
I implemented it this way the first time, but after realizing that the CLI always drops the connection after proving, it didn't make a lot of sense. But maybe for different uses this would be a useful feature (maybe the web client would benefit from it).

Reply:
Yeah - unless it adds much complexity, I'd "cache" the connection because there may be other "long-living" client implementations.